tripy/examples/resent50/resnet50.ipynb

```diff
@@ -0,0 +1,529 @@
+    "        if 'classifier' in k:\n",
+    "            k = k.replace('1.', '') # Remove extra 1 (slight naming mismtach)\n",
+    "\n",
+    "        assert k in set(model_dict.keys())\n",
+    "        new_v = tp.Parameter(v.contiguous())\n",
+    "        model_dict[k] = new_v \n",
+    "\n",
+    "    model.load_state_dict(model_dict)\n",
+    "    return model\n",
+    "\n",
+    "pretrained_state_dict = resnet_pretrained.state_dict()\n",
+    "tripy_model = load_pretrained_weights_tp(tripy_model, pretrained_state_dict)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Now we compile the model with static input shape"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "input_shape = [1, 3, 224, 224]\n",
+    "compiled_model = tp.compile(tripy_model, args=[tp.InputInfo(input_shape, dtype=tp.float32)])"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Run a dummy forward path\n",
+    "x = tp.ones(input_shape, dtype=tp.float32)\n",
+    "compiled_output = compiled_model(x)\n",
+    "compiled_output.shape"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Tripy Demo\n",
+    "\n",
+    "Now we will test our tripy resnet50 classification model on a few sample images for image classification task "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Prepare the image for Tripy\n",
+    "processed_image = processor(image, return_tensors=\"np\")['pixel_values']\n",
+    "tp_image = tp.Tensor(processed_image, dtype=tp.float32, device=tp.device(\"gpu\"))\n",
+    "tp_image.shape"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Run the image through the model\n",
+    "tp_logits = compiled_model(tp_image)\n",
+    "tp_logits.shape"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Get the predicted label\n",
+    "tp_predicted_label = torch.argmax(torch.from_dlpack(tp_logits), -1).tolist()[0]\n",
+    "print(f\"Predicted Label: {resnet_pretrained.config.id2label[tp_predicted_label]}\")\n",
+    "image"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 18,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def get_tp_pred(model, image, processor):\n",
+    "    processed_image = processor(image, return_tensors=\"np\")['pixel_values']\n",
+    "    tp_image = tp.Tensor(processed_image, dtype=tp.float32, device=tp.device(\"gpu\"))\n",
+    "    tp_logits = model(tp_image)\n",
+    "    return torch.argmax(torch.from_dlpack(tp_logits), -1).tolist()[0]\n",
+    "\n",
+    "def get_exact_answer(model, image, processor):\n",
```

Comment:
Can we rename to `get_torch_pred`?

Reply:
Hmm, I can remove it if you just want to do asserts. It was more to demonstrate that the trophy model works the same as the torch model since they both generate the same label. But I guess I did not communicate that well. 

Question is,
1- do we want to keep the cross comparison with torch model in this notebook? or do we just use torch model weights and run the demo on Tripy and let the user figure out that the labels for the pictures are reasonable?
2- letting the user figure out maybe ambiguous though, since If I am a user, I would personally be curious to see if the tripy model performs the same as torch model? How can I see that without that label assertion/print? (this is completely subjective though, since it is a tripy example demo, it is probably not necessary...)

Reply:
The key point is that the model works as its expected to (i.e. is able to correctly classify images). Comparing against torch is just a proxy for that, and if we already know what labels we expect, we can check directly against those instead of introducing more complexity.

Reply:
Sounds good, added hard coded labels!